src/service/video-manager-impl.js

```diff
@@ -436,6 +540,136 @@ class VideoEntry {
   }
 
   /**
+   * Called when the video's position in the viewport changed.
+   * @param {PositionInViewportEntryDef} newPos
+   * @private
+   */
+  onPositionChanged_(newPos) {
+
+    const vidElement = scopedQuerySelector(this.video.element, 'video, iframe');
+    const vidStyle = vidElement.style;
+
+    // How much to scale the video by when minimized
+    const SCALE = 0.6;
+
+    // Temporary fix until PositionObserver somehow tracks objects outside of
+    // the viewport
+    if (!newPos.positionRect
+       && this.getPlayingState() == PlayingStates.PLAYING_MANUAL
+       && this.minimizePosition_ != MinimizePositions.DEFAULT) {
+      // Hide the controls.
+      this.video.hideControls();
+      vidElement.classList.add('i-amphtml-dockable-video-minimizing');
+      vidStyle.height = SCALE * this.initialHeight_ + 'px';
+      vidStyle.width = SCALE * this.initialWidth_ + 'px';
+      vidStyle.borderRadius = '6px';
+      vidStyle.left = '20px';
+      vidStyle.boxShadow = '0px 0px 20px 0px rgba(0, 0, 0, 0.3)';
+      if (this.minimizePosition_ == MinimizePositions.BOTTOM) {
+        this.minimizePosition_ = MinimizePositions.TOP;
+        vidStyle.bottom = '20px';
+        vidStyle.top = 'auto';
+      } else {
+        this.minimizePosition_ = MinimizePositions.BOTTOM;
+        vidStyle.top = '20px';
+        vidStyle.bottom = 'auto';
+      }
+      return;
+    }
+
+    const docViewTop = newPos.viewportRect.top;
```

Comment:
(skipping review for this part since switching to using `scale` will likely change a lot of the code here,)